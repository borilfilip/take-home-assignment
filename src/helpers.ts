export const apiUrl = 'http://localhost:8000';

interface Address {
  city: string;
  street: string;
  number: number;
}

export interface BookingObject {
  id?: string;
  start: Address;
  destination: Address;
  when: string;
  people: number;
}

export interface HTTPValidationError {
  detail: { loc: string; msg: string; type: string }[];
}
