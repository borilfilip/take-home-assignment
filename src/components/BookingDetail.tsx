import { BookingObject } from '../helpers';

interface BookingDetailProps {
  booking?: BookingObject;
}

function BookingDetail({ booking }: BookingDetailProps) {
  return (
    (booking && (
      <>
        <div className="mb-3">
          <strong>Start</strong>
          <br />
          {booking.start.street} {booking.start.number},<br />
          {booking.start.city}
        </div>
        <div className="mb-3">
          <strong>Destination</strong>
          <br />
          {booking.destination.street} {booking.destination.number},<br />
          {booking.destination.city}
        </div>
        <div className="mb-3">
          <strong>Date</strong>
          <br />
          {new Date(booking.when).toLocaleString()}
        </div>
        <div className="mb-3">
          <strong>Number of people</strong>
          <br />
          {booking.people}
        </div>
      </>
    )) ||
    null
  );
}

export default BookingDetail;
