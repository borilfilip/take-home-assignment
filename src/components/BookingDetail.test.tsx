import renderer from 'react-test-renderer';
import { BookingObject } from '../helpers';
import BookingDetail from './BookingDetail';

export const mockBooking: BookingObject = {
  id: '1',
  start: {
    city: 'Warszaw',
    street: 'Marszałkowska',
    number: 1
  },
  destination: {
    city: 'Praha',
    street: 'Staroměstské nám.',
    number: 4
  },
  when: '2022-12-08T08:15:37.164Z',
  people: 2
};

it('renders booking detail', () => {
  const component = renderer.create(<BookingDetail booking={mockBooking} />);
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
