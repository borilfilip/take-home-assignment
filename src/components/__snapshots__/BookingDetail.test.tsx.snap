// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders booking detail 1`] = `
Array [
  <div
    className="mb-3"
  >
    <strong>
      Start
    </strong>
    <br />
    Marszałkowska
     
    1
    ,
    <br />
    Warszaw
  </div>,
  <div
    className="mb-3"
  >
    <strong>
      Destination
    </strong>
    <br />
    Staroměstské nám.
     
    4
    ,
    <br />
    Praha
  </div>,
  <div
    className="mb-3"
  >
    <strong>
      Date
    </strong>
    <br />
    12/8/2022, 7:15:37 PM
  </div>,
  <div
    className="mb-3"
  >
    <strong>
      Number of people
    </strong>
    <br />
    2
  </div>,
]
`;
